langchain/src/llms/bedrock-chat.ts

```diff
@@ -0,0 +1,267 @@
+  the generated text from the model response. */
+
+  static prepareInput(provider: string, prompt: string): Dict {
+    const inputBody: Dict = {};
+
+    if (provider === "anthropic" || provider === "ai21") {
+      inputBody.prompt = prompt;
+    } else if (provider === "amazon") {
+      inputBody.inputText = prompt;
+      inputBody.textGenerationConfig = {};
+    } else {
+      inputBody.inputText = prompt;
+    }
+
+    if (provider === "anthropic" && !("max_tokens_to_sample" in inputBody)) {
+      inputBody.max_tokens_to_sample = 50;
+    }
+
+    return inputBody;
+  }
+
+  static prepareOutput(
+    provider: string,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    responseBody: any,
+    streaming: boolean
+  ): string {
+    if (streaming) {
+      if (provider === "anthropic") {
+        return responseBody.completion;
+      } else if (provider === "ai21") {
+        return responseBody.data.text;
+      }
+      return responseBody.outputText;
+    } else {
+      if (provider === "anthropic") {
+        return responseBody.completion;
+      } else if (provider === "ai21") {
+        return responseBody.completions[0].data.text;
+      }
+      return responseBody.results[0].outputText;
+    }
+  }
+}
+
+/** Bedrock models.
+    To authenticate, the AWS client uses the following methods to automatically load credentials:
+    https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html
+    If a specific credential profile should be used, you must pass the name of the profile from the ~/.aws/credentials file that is to be used.
+    Make sure the credentials / roles used have the required policies to access the Bedrock service.
+*/
+export interface BedrockInput {
+  /** Model to use.
+      For example, "amazon.titan-tg1-large", this is equivalent to the modelId property in the list-foundation-models api.
+  */
+  model: string;
+
+  /** The AWS region e.g. `us-west-2`.
+      Fallback to AWS_DEFAULT_REGION env variable or region specified in ~/.aws/config in case it is not provided here.
+  */
+  region?: string;
+
+  /** AWS Credentials.
+      If no credentials are provided, the default credentials from `@aws-sdk/credential-provider-node` will be used.
+   */
+  credentials?: CredentialType;
+
+  /** Temperature */
+  temperature?: number;
+
+  /** Max tokens */
+  maxTokens?: number;
+
+  /** Streaming */
+  streaming?: boolean;
+
+  /** A custom fetch function for low-level access to AWS API. Defaults to fetch() */
+  fetchFn?: typeof fetch;
+}
+
+export class BedrockChat extends LLM implements BedrockInput {
+  model = "amazon.titan-tg1-large";
+
+  region: string;
+
+  credentials: CredentialType;
+
+  temperature?: number | undefined = undefined;
+
+  maxTokens?: number | undefined = undefined;
+
+  streaming?: boolean | undefined = false;
+
+  fetchFn: typeof fetch;
+
+  marshaller: EventStreamMarshaller = new EventStreamMarshaller(
```

Comment:
I think this has been deprecated?
<img width="876" alt="Screenshot 2023-08-22 at 7 53 14 PM" src="https://github.com/hwchase17/langchainjs/assets/6952323/21fec63c-a275-4309-b3c4-d9d52a7d4774">

Reply:
Yes, sorry I missed that.  I will replace with a suitable alternative for this 👍